File: clients/android/Vertex/app/src/main/kotlin/ru/vertices/android/ui/components/VxProtocolBadge.kt
package ru.vertices.android.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import ru.vertices.android.ui.theme.LocalVertexColors
import ru.vertices.android.ui.theme.VxCaptionMonoStyle

/**
 * Compact protocol chip used in Settings → Active Configuration.
 * Mirror of `VxProtocolBadge.swift`. Caller passes label uppercased.
 */
@Composable
fun VxProtocolBadge(label: String, isPrimary: Boolean, modifier: Modifier = Modifier) {
    val tokens = LocalVertexColors.current
    Text(
        text = label,
        style = VxCaptionMonoStyle,
        color = if (isPrimary) tokens.accentPrimary else tokens.textTertiary,
        modifier = modifier
            .clip(RoundedCornerShape(4.dp))
            .background(tokens.bgSurfaceMuted)
            .padding(horizontal = 6.dp, vertical = 2.dp),
    )
}

File: clients/ios/Vertex/App/Views/Components/VxProtocolBadge.swift
import SwiftUI

/// Compact protocol chip used in Settings → Active Configuration.
/// Caller must pass the label already uppercased (e.g. "MQTTS", "WSS").
struct VxProtocolBadge: View {
    let label: String
    let isPrimary: Bool

    var body: some View {
        Text(label)
            .font(.vxCaptionMono)
            .foregroundStyle(isPrimary ? Color.accentPrimary : Color.textTertiary)
            .padding(.horizontal, 6)
            .padding(.vertical, 2)
            .background(Color.bgSurfaceMuted, in: .rect(cornerRadius: 4))
    }
}
